lib/boxcars.rb

```diff
@@ -62,7 +62,7 @@ def serpapi_api_key(**kwargs)
 
     def check_key(key, val)
       unless val.nil? || val.empty?
-        instance_variable_set("@#{key}", val)
+        instance_variable_set("@#{key}", val) unless instance_variable_get("@#{key}")
```

Comment:
If you look at my previous comment, I don't think the value should be saved here.